pkg/operator2/datasync/syncdata.go

```diff
@@ -0,0 +1,238 @@
+package datasync
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corelistersv1 "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+// data maps dest -> source
+// dest is metadata.name for resource in our deployment's namespace
+type ConfigSyncData struct {
+	data map[string]sourceData
+}
+
+type ResourceType string
+
+const (
+	ConfigMapType ResourceType = "configMap"
+	SecretType    ResourceType = "secret"
+)
+
+type sourceData struct {
+	Src      string       `json:"src"`  // name of the source in openshift-config namespace
```

Comment:
nit: why isn't the name of this field just `Name` ? As per further comments on fields in this struct, a good name can largely obviate the need for explanatory comments.

Reply:
You're right, I was blindly refactoring the old code without thinking about the names too much, but these name suggestion do make the code more readable, addressed :+1: